tee-worker/app-libs/stf/src/helpers.rs

```diff
@@ -112,23 +112,30 @@ pub fn set_block_number(block_number: u32) {
 
 pub fn is_authorized_signer<AccountId: Encode + Decode + PartialEq>(
 	signer: &AccountId,
-	who: &AccountId,
+	who: Option<AccountId>,
 ) -> bool {
-	signer == &enclave_signer_account::<AccountId>() || signer == who
+	match who {
+		Some(ref who) => signer == &enclave_signer_account::<AccountId>() || signer == who,
+		None => false,
+	}
 }
 
 // verification message format:
 // blake2_256(<sidechain nonce> + shieldingKey.encrypt(<primary account> + <identity-to-be-linked>).ciphertext)
 // where <> means SCALE-encoded
 // see https://github.com/litentry/litentry-parachain/issues/1739
-pub fn get_expected_raw_message<AccountId: Encode + Decode>(
-	who: &AccountId,
+pub fn get_expected_raw_message(
+	who: &Identity,
 	identity: &Identity,
 	sidechain_nonce: Index,
 	key: UserShieldingKeyType,
 	nonce: UserShieldingKeyNonceType,
 ) -> Vec<u8> {
-	let mut data = who.encode();
+	let mut data = match who {
```

Comment:
Now that `who` is `Identity` too, shall we replace the encoded _address_ with encoded _Identity_? People might wonder why you encode the whole to-be-linked identity but only the address for the prime identity.

Apart from that, one argument is we _might_ allow web2 login in the future, so trying to be a bit future compatible here.

It would be a breaking change for clients, but we have enough of that already :)

Tests need to be adjusted if we go for it - or if you like more discussions we'd better have a separate PR for it.

Reply:
Changed